tbx/project_styleguide/templates/patterns/molecules/mode_switcher/mode_switcher.html

```diff
@@ -0,0 +1,35 @@
+<form class="mode-switcher" data-mode-switcher method="GET" action="{% url 'switch_mode' %}">
+    <fieldset class="mode-switcher__layout">
+        <legend class="sr-only">Toggle light or dark mode</legend>
+        <label for="id-mode-switcher-light{% if mobile %}-mobile{% endif %}" class="mode-switcher__label mode-switcher__label--light" aria-label="Switch to light mode">
+            {% include "patterns/atoms/icons/icon.html" with name="light-mode-icon" classname="mode-switcher__icon mode-switcher__icon--light" %}
+        </label>
+        <div class="mode-switcher__button">
+            <input
+                type="radio"
+                name="switch_mode"
+                value="light"
+                class="mode-switcher__radio mode-switcher__radio--light"
+                id="id-mode-switcher-light{% if mobile %}-mobile{% endif %}"
+                data-mode-switch
+                {% if MODE == 'light' %}checked="checked"{% endif %}
+            >
+            <input
+                type="radio"
+                name="switch_mode"
+                value="dark"
+                class="mode-switcher__radio mode-switcher__radio--dark"
+                id="id-mode-switcher-dark{% if mobile %}-mobile{% endif %}"
+                data-mode-switch
+                {% if MODE == 'dark' %}checked="checked"{% endif %}
```

Comment:
The mobile toggle doesn't look like it's respecting the users preference - if I choose dark mode and refresh the page the toggle is showing that light mode is still selected. 

<details><summary>Example video</summary>
<p>


https://github.com/user-attachments/assets/200b6468-10b1-44d6-9c35-be3370696849



</p>
</details>

Reply:
I looked into this and it looks like it's because of the `{% wagtailcache 600 "primarynavmobile" current_site.pk is_pattern_library %}` tag. If you move it underneath `{% endwagtailcache %}` it works as expected.

Reply:
I don't think this should be an issue. The cache applies to the template code and I think you had probably just loaded a previous version of that which had got cached. But I'll maybe move it outside the cache just to be sure.